src/server/pps/server/api_server.go

```diff
@@ -4139,3 +4139,64 @@ func (a *apiServer) SetProjectDefaults(ctx context.Context, req *pps.SetProjectD
 	}
 	return &resp, nil
 }
+
+// PipelinesSummary implements the protobuf pps.PipelinesSummary RPC
+func (a *apiServer) PipelinesSummary(ctx context.Context, req *pps.PipelinesSummaryRequest) (*pps.PipelinesSummaryResponse, error) {
+	var projects []*pfs.Project
+	for _, p := range req.Projects {
+		var project *pfs.Project
+		switch p.Picker.(type) {
+		case *pfs.ProjectPicker_Name:
+			project = &pfs.Project{Name: p.GetName()}
+		default:
+			return nil, errors.Errorf("project picker is of an unknown type: %T", p.Picker)
+		}
+		if err := a.env.AuthServer.CheckProjectIsAuthorized(ctx, project, auth.Permission_PROJECT_LIST_REPO); err != nil {
+			return nil, errors.Wrapf(err, "not authorized to list repos of project %q", project.String())
+		}
+		projects = append(projects, project)
+	}
+	projectsMap := make(map[string]struct{})
+	for _, p := range projects {
+		projectsMap[p.String()] = struct{}{}
+	}
+	summaries := make(map[string]*pps.PipelinesSummary)
+	resp := &pps.PipelinesSummaryResponse{}
+	if err := ppsutil.ListPipelineInfo(ctx, a.pipelines, nil, 0, func(pi *pps.PipelineInfo) error {
```

Comment:
Pity that we haven’t relationalized PPS yet … this could probably be a single DB query.

Reply:
totally